test/e2e/statefulset.go

```diff
@@ -315,16 +315,16 @@ var _ = framework.KubeDescribe("StatefulSet", func() {
 			expectedPodName := ss.Name + "-1"
 			expectedPod, err := f.ClientSet.Core().Pods(ns).Get(expectedPodName, metav1.GetOptions{})
 			Expect(err).NotTo(HaveOccurred())
+
+			By("Verifying the 2nd pod is removed only when it becomes running and ready")
+			sst.RestoreProbe(ss, testProbe)
```

Comment:
This calls `RunHostCmd` which is a simple exec into the pod though, no real host command as it was long ago. So this shouldn't take long (only does a file move). Hence, I guess this change doesn't make much of a difference. The issue we see is that `watch.Until` can return with `Keep waiting, received error from watch event: too old resource version: 7504 (7561)`.